firstvoices/backend/resources/utils/import_export_widgets.py

```diff
@@ -105,13 +105,26 @@ def __init__(self, *args, **kwargs):
 
     def clean(self, value, row=None, **kwargs):
         category_match_pattern = r"category[_2-5]*"
-        input_categories = [
-            value
-            for key, value in row.items()
-            if re.fullmatch(category_match_pattern, key)
-        ]
 
+        input_categories = {}
+        valid_categories = []
+
+        for column, input_value in row.items():
+            if re.fullmatch(category_match_pattern, column):
+                input_categories[column] = input_value
+
+        # If no categories provided, return
         if len(input_categories) == 0:
             return Category.objects.none()
 
-        return Category.objects.filter(title__in=input_categories)
+        # Validate categories
+        for column, input_value in input_categories.items():
+            category_lookup = Category.objects.filter(
+                site__id=row["site"], title=input_value
+            )
+            if len(category_lookup) == 0:
+                raise ValidationError(f"Invalid category supplied in column {column}.")
```

Comment:
Just to make sure I understand how these errors work-- this will lead to this row being skipped in the import but will not fail the whole import, right?

Reply:
Yes, that's right. And this error message will go to this row's errors list.